idb/dummy.go

```diff
@@ -233,6 +238,7 @@ type IndexerDb interface {
 	CommitRoundAccounting(updates RoundUpdates, round uint64, blockPtr *types.Block) (err error)
 
 	GetBlock(round uint64) (block types.Block, err error)
+	BlockWithTransactions(ctx context.Context, round uint64, tf TransactionFilter) (block types.Block, transactions []TxnRow, err error)
```

Comment:
If we add this feature to the interface I would probably merge this into `GetBlock`, adding in a flag (or "Options" object) to request that transactions be included in the result.

Also the TransactionFilter object isn't required, inside the function we would initialize the `TransactionFilter{Round:round}`

Reply:
If we merge these two functions, the rest of my feedback probably no longer applies.